dataflow/src/org/checkerframework/dataflow/analysis/Store.java

```diff
@@ -54,6 +54,25 @@
     S leastUpperBound(S other);
 
     /**
+     * Compute an upper bound of two stores that is wider than the least upper bound of the two
+     * stores.
+     *
+     * <p>(This method is optional, if not required, implementations should call leastUpperBound.)
+     *
+     * <p><em>Important</em>: This method must fulfill the following contract:
+     *
+     * <ul>
+     *   <li>Does not change {@code this}.
+     *   <li>Does not change {@code other}.
+     *   <li>Returns a fresh object which is not aliased yet.
+     *   <li>Returns an object of the same (dynamic) type as {@code this}, even if the signature is
```

Comment:
I guessed method signature, too, but that doesn't really make sense: it first says that it has to have the same dynamic type. If it's the same type, it has exactly the same method signatures.